gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -886,6 +892,21 @@ public JsonWriter newJsonWriter(Writer writer) throws IOException {
     if (prettyPrinting) {
       jsonWriter.setIndent("  ");
     }
+    switch (newlineStyle) {
+      case CURRENT_OS:
+        jsonWriter.setNewline(System.lineSeparator());
+        break;
+      case WINDOWS:
+        jsonWriter.setNewline("\r\n");
+        break;
+      case OLD_MACOS:
+        jsonWriter.setNewline("\r");
+        break;
+      case MACOS_AND_LINUX: // intentional fallthrough
+      default:
```

Comment:
Because this `switch` statement covers all constants, it might be good to omit the `default` case to get IDE warnings if a constant is not covered (not sure though if `javac` or CodeQL also warn about this), or use `throw new AssertionError()`.

Reply:
I've checked to see who complains about an enum value not covered by a switch, and:
* just `javac` (no options) does not complain
* the regular `maven install` does not complain
* Visual Studio Code does not complain
* IntelliJ only complains if you run a code analyze, but not for a regular build
* Eclipse gives a warning, out of the box, there is no need to configure anything

So I would rather keep this "as is", if you don't mind.

Reply:
Thanks for testing this so extensively!

What do you think about `default: throw new AssertionError()`? My concern is that the `default` case is really not supposed to be reached.

Reply:
Done.